Tasks/KubernetesManifestV0/src/utils/KubernetesObjectUtility.ts

```diff
@@ -287,6 +287,53 @@ function setImagePullSecrets(inputObject: any, newImagePullSecrets: any) {
     return;
 }
 
+export function updateImageDetails(inputObject: any, containers: string[]) {
+    if (!inputObject || !inputObject.spec || !containers) {
+        return;
+    }
+
+    if (inputObject.spec.template && !!inputObject.spec.template.spec) {
+        if (inputObject.spec.template.spec.containers) {
+            updateContainers(inputObject.spec.template.spec.containers, containers);
+        }
+        if (inputObject.spec.template.spec.initContainers) {
+            updateContainers(inputObject.spec.template.spec.initContainers, containers);
+        }
+        return;
+    }
+
+    if (inputObject.spec.containers) {
+        updateContainers(inputObject.spec.containers, containers);
+    }
+
+    if (inputObject.spec.initContainers) {
+        updateContainers(inputObject.spec.initContainers, containers);
+    }
+}
+
+function updateContainers(containers: any[], images: string[]) {
+    if (!containers || containers.length === 0) {
+        return containers;
+    }
+    containers.forEach((container) => {
+        const imageName: string = container.image.trim();
+        let img = '';
+        if (imageName.indexOf('/') > 0) {
+            const imgParts = imageName.split('/');
+            const registry = imgParts[0];
+            const imgName = imgParts[1].split(':')[0];
+            img = `${registry}/${imgName}`;
+        } else {
+            img = imageName.split(':')[0];
+        }
+        images.forEach(image => {
+            if (image.startsWith(img + ':') || image === img) {
```

Comment:
if it is same then no need to replace right. image===img we can remove.

Reply:
Good point. The check should be different. I'll fix it